api/app/models/env_canada.py

```diff
@@ -535,19 +541,28 @@ def main():
     # check if we encountered any exceptions.
     if env_canada.exception_count > 0:
         # if there were any exceptions, return a non-zero status.
-        logger.warning('completed processing with some exceptions')
-        sys.exit(os.EX_SOFTWARE)
+        raise CompletedWithSomeExceptions()
     return env_canada.files_processed
 
 
-if __name__ == "__main__":
+def main():
+    """ main script - process and download models, then do exception handling """
     try:
-        main()
-    # pylint: disable=broad-except
-    except Exception as exception:
+        process_models()
+    except CompletedWithSomeExceptions:
+        logger.warning('completed processing with some exceptions')
+        sys.exit(os.EX_SOFTWARE)
+    except Exception as exception:  # pylint: disable=broad-except
         # We catch and log any exceptions we may have missed.
         logger.error('unexpected exception processing', exc_info=exception)
+        rc_message = ':poop: Encountered error retrieving model data from Env Canada\n{}: {}'.format(
```

Comment:
I think it would be better if you concatenate the hostname and exception inside of `send_rocketchat_notification`. And you pass the first part of the message and the exception obj as arguments